src/transformStateWithClones.js

```diff
@@ -7,7 +7,28 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  const arrayResult = [];
+  let result = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      result = Object.assign(result, action.extraData);
+    }
+
+    if (action.type === 'removeProperties') {
+      for (const key of action.keysToRemove) {
+        delete result[key];
+      }
+    }
 
+    if (action.type === 'clear') {
+      for (const clear in result) {
+        delete result[clear];
+      }
+    }
+    arrayResult.push({ ...result });
```

Comment:
[CODE STYLE] - don't mutate object or arrays - it will cause unexpected results later on. Instead of mutating the `result` object directly, you should create a new copy for each action to ensure that previous states in `arrayResult` are not affected by subsequent actions.